code/cgame/cg_event.c

```diff
@@ -2342,7 +2374,7 @@ void CG_EntityEvent( centity_t *cent, vec3_t position ) {
 	case EV_WEAPON_CHARGE:
 		DEBUGNAME("EV_WEAPON_CHARGE");
 		assert(es->eventParm > WP_NONE && es->eventParm < WP_NUM_WEAPONS);
-		if (VALID_INDEX(cg_weapons, es->eventParm) && cg_weapons[es->eventParm].chargeSound)
+		if ((es->eventParm > WP_NONE && es->eventParm < WP_NUM_WEAPONS) && cg_weapons[es->eventParm].chargeSound)
```

Comment:
Wouldn't it be more correct to check `VALID_INDEX(cg_weapons, es->eventParm) && cg_weapons[es->eventParm].registered`? Don't see what's the problem with original exactly (other than potentially playing 0 sound)